features/steps/test_starter/{{ cookiecutter.repo_name }}/src/{{ cookiecutter.python_package }}/pipeline_registry.py

```diff
@@ -11,7 +11,7 @@ def register_pipelines() -> Dict[str, Pipeline]:
     """Register the project's pipelines.
 
     Returns:
-        A mapping from a pipeline name to a ``Pipeline`` object.
+        A mapping from pipeline names to ``Pipeline`` objects.
```

Comment:
@AntonyMilneQB What would the resulting registry look like?

```python
    pipelines = find_pipelines()
    pipelines["__default__"] = sum(pipelines.values())
    pipelines["data_processing"] = pipeline(
        pipelines["data_engineering"], namespace="data_processing"
    )
    return pipelines
```

And then update the references to use the full "data_engineering", "data_science", "data_processing" instead of "de", "ds", "dp", respectively?

Reply:
If possible I'd like the spaceflights tutorial to work so that you don't have to touch `pipeline_registry.py` at all. I believe this should be possible until this point anyway: https://kedro.readthedocs.io/en/stable/tutorial/namespace_pipelines.html

Currently we do `kedro pipeline create data_processing` and `data_science` and then alias them as `dp` and `ds`. There's no namespacing involved in the spaceflights starter. I think the only thing that should change is that you need to specify the full pipeline name when using `kedro run -p` rather than the alias.

Hence the resulting registry used in the spaceflights starter and in this test starter should just be the same as the default `kedro new` template.

Reply:
I managed to get some manual testing done:

1. Standard spaceflights worked without any problems, pipeline was registered correctly.
2. Having no pipelines gave an exception*, only default pipeline was registered.
3. Having no node.py in a pipeline's dir + empty pipeline.py gave an exception, pipeline still registered.
4. Having no pipeline.py in a pipeline's dir gave an exception, pipeline not registered.
5. Having empty node.py / pipeline.py gave an exception, pipeline registered.

* In every case, the exception I got was: `ValueError: Pipeline contains no nodes after applying all provided filters`, which is descriptive enough in my view.

Reply:
@AntonyMilneQB I've updated this to use `find_pipelines`!

Also, thanks @jmholzer for testing all those scenarios. ❤️